airflow/jobs/scheduler_job.py

```diff
@@ -816,7 +811,12 @@ def process_file(self, file_path, zombies, pickle_dags=False, session=None):
         # Save individual DAGs in the ORM and update DagModel.last_scheduled_time
         dagbag.sync_to_db()
 
-        paused_dag_ids = {dag.dag_id for dag in dagbag.dags.values() if dag.is_paused}
+        paused_dag_ids = (
+            session.query(DagModel.dag_id)
+            .filter(DagModel.is_paused.is_(True))
+            .filter(DagModel.dag_id.in_(dagbag.dag_ids))
```

Comment:
Is IN query recommended here for use cases where we have 10 thousands of paused dag? Shouldn't query be broken into smaller batches?

Reply:
This is executed only for DAGs from one file. You would have to have a few thousand DAGs in one file. For now, I focused on the situation when we have up to 200 DAGs. If we want to support several thousand DAGs in one file, we need to introduce much more optimization and this one would not change anything.

Reply:
Sorry for trouble, I missed **DAGs from one file** part.